cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/DataPipelineApp.java

```diff
@@ -60,6 +61,7 @@ public void configure() {
     // if this is for the system services and not an actual pipeline
     if (config.isService()) {
       addService(new StudioService());
+      addService(new ConnectionService());
```

Comment:
does it need to be a new Service or can it be just a new handler? (new service would require at least one more container)

Reply:
Initially I thought it is good to have a new service for it to be clearer on service responsibility. But considering adding a new container will take up more resources, I have changed it be inside the studio service